### PR TITLE
выполнение домашики 3

### PR DESCRIPTION
**Pull request by itreyser**

### CODE REVIEW

File: src/lesson3/SingleArray.java [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        Scanner scan = new Scanner(System.in);
+        System.out.println(Arrays.toString(array));
+        System.out.println("Введите число, которое хотите найти в массиве: ");
+        int input = scan.nextInt();
+        for (int i = 0; i < array.length; i++) {
+            if (array[i] == input) {
+                System.out.println("Ваше число найдено в массиве!");
+                break;
+            } else if (i == array.length - 1) {
+                System.out.println("Ваше число не найдено в массиве!");
```
Comment (DHrytsuk) [inline src/lesson3/SingleArray.java]: This should be out of loop

File: src/lesson3/SingleArray.java [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        int[] array = new int[20];
+        Random r = new Random();
+        for (int i = 0; i < array.length; i++) {
+            array[i] = r.nextInt();
+        }
+        Scanner scan = new Scanner(System.in);
+        System.out.println(Arrays.toString(array));
+        System.out.println("Введите число, которое хотите удалить в массиве: ");
+        int input = scan.nextInt();
+        ArrayList<Integer> list = new ArrayList<>();
```
Comment (DHrytsuk) [inline src/lesson3/SingleArray.java]: You should resolve it without collections

File: src/lesson3/SingleArray.java [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+        double[] array = new double[scan.nextInt()];
+        for (int i = 0; i < array.length; i++) {
+            array[i] = Math.random();
+        }
+        double min = array[0];
+        double max = array[0];
+        double counter = 0;
+        for (int i = 0; i < array.length; i++) {
+            counter += array[i];
+            if (min < array[i]) {
```
Comment (DHrytsuk) [inline src/lesson3/SingleArray.java]: Is it correct?)

File: src/lesson3/SingleArray.java [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+        int[] array2 = new int[5];
+        Random r = new Random();
+        for (int i = 0; i < 5; i++) {
+            array1[i] = r.nextInt();
+            array2[i] = r.nextInt();
+        }
+        double counter1 = 0;
+        double counter2 = 0;
+        for (int i = 0; i < 5; i++) {
+            counter1 = array1[i];
```
Comment (DHrytsuk) [inline src/lesson3/SingleArray.java]: ?

File: src/lesson3/SingleArray.java [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+            System.out.println("Размер массива должен быть от 6 и до 10 включительно");
+            size = scan.nextInt();
+        }
+        int[] array = new int[size]; // создание массива
+        Random r = new Random();
+        for (int i = 0; i < array.length; i++) { // заполнение массива рандомом
+            array[i] = r.nextInt();
+        }
+        System.out.println(Arrays.toString(array));
+        ArrayList<Integer> list = new ArrayList<>();
```
Comment (DHrytsuk) [inline src/lesson3/SingleArray.java]: It should be resolved without collections

File: src/lesson3/SingleArray.java [hunk 6]
```diff
@@ -38,23 +42,29 @@ public static void removeInArray() {
...
         System.out.println("Введите число, которое хотите удалить в массиве: ");
         int input = scan.nextInt();
-        ArrayList<Integer> list = new ArrayList<>();
-        if ((Arrays.binarySearch(array, input)) >= 0) {
-            System.out.println("Такого значения нету в массиве!");
+        boolean found = false;
+        for (int i = 0; i < array.length; i++) {
+            if (array[i] == input) {
+                found = true;
+                array[i] = 0; // помечаем элемент на удаление
```
Comment (DHrytsuk) [inline src/lesson3/SingleArray.java]: The unutual array shouldn't be modified

File: src/lesson3/SingleArray.java [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+                break;
+            }
+        }
+        if (!found) {
+            System.out.println("Такое число не найдено в массиве.");
+        } else {
+            int count = 0;
+            for (int i = 0; i < array.length; i++) {
+                if (array[i] != input) {
+                    array[count] = array[i];
```
Comment (DHrytsuk) [inline src/lesson3/SingleArray.java]: You still modify initial array

File: src/lesson3/SingleArray.java [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+        Scanner scan = new Scanner(System.in);
+        System.out.println(Arrays.toString(array));
+        System.out.println("Введите число, которое хотите удалить в массиве: ");
+        int input = scan.nextInt();
+        boolean found = false;
+        int countInput = 0;
+        for (int j : array) {
+            if (j == input) {
+                found = true;
+                countInput += 1;
```
Comment (DHrytsuk) [inline src/lesson3/SingleArray.java]:
```suggestion
                countInput ++;
```

File: src/lesson3/SingleArray.java [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+        int[] array = new int[20];
+        Random r = new Random();
+        for (int i = 0; i < array.length; i++) {
+            array[i] = r.nextInt();
+        }
+        Scanner scan = new Scanner(System.in);
+        System.out.println(Arrays.toString(array));
+        System.out.println("Введите число, которое хотите удалить в массиве: ");
+        int input = scan.nextInt();
+        boolean found = false;
```
Comment (DHrytsuk) [inline src/lesson3/SingleArray.java]: Maybe, if you have `countInput`, which determines, whether we have necessary number, we can remove this flag?

Comment (itreyser) [reply src/lesson3/SingleArray.java]: my bad

**Review by DHrytsuk — APPROVED** [review]

**Pull request merged**